crates/optimism/consensus/src/validation/mod.rs

```diff
@@ -48,31 +47,13 @@ where
         ))
     }
 
-    match (header.withdrawals_root(), body.calculate_withdrawals_root()) {
-        (Some(header_withdrawals_root), Some(withdrawals_root)) => {
-            // after isthmus, the withdrawals root field is repurposed and no longer mirrors the
-            // withdrawals root computed from the body
-            if chain_spec.is_isthmus_active_at_timestamp(header.timestamp()) {
-                // After isthmus we only ensure that the body has empty withdrawals
-                if withdrawals_root != EMPTY_ROOT_HASH {
-                    return Err(ConsensusError::BodyWithdrawalsRootDiff(
-                        GotExpected { got: withdrawals_root, expected: EMPTY_ROOT_HASH }.into(),
-                    ))
-                }
-            } else {
-                // before isthmus we ensure that the header root matches the body
-                if withdrawals_root != header_withdrawals_root {
-                    return Err(ConsensusError::BodyWithdrawalsRootDiff(
-                        GotExpected { got: withdrawals_root, expected: header_withdrawals_root }
-                            .into(),
-                    ))
-                }
-            }
-        }
-        (None, None) => {
-            // this is ok because we assume the fork is not active in this case
-        }
-        _ => return Err(ConsensusError::WithdrawalsRootUnexpected),
+    if chain_spec.is_isthmus_active_at_timestamp(header.timestamp()) {
+        // After isthmus we only ensure that the body has empty withdrawals
+        isthmus::ensure_withdrawals_storage_root_is_some(header)
+            .map_err(|err| ConsensusError::Other(err.to_string()))?;
+    } else {
```

Comment:
```suggestion
    } else {
```
this needs to be an else if checking if canyon is active